src/Container/AgFeatureGrid/AgFeatureGrid.jsx

```diff
@@ -0,0 +1,1021 @@
+    const {
+      map,
+      selectable
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    map.un('pointermove', this.onMapPointerMove);
+
+    if (selectable) {
+      map.un('singleclick', this.onMapSingleClick);
+    }
+  }
+
+  /**
+   * Returns the column definitions out of the attributes of the first
+   * given feature.
+   *
+   * @return {Array} The column definitions.
+   */
+  getColumnDefs = () => {
+    const {
+      attributeBlacklist,
+      features,
+      columnDefs,
+      selectable
+    } = this.props;
+
+    let columns = [];
+    const feature = features[0];
+
+    if (!(feature instanceof OlFeature)) {
+      return columns;
+    }
+
+    const props = feature.getProperties();
+
+    if (selectable) {
+      columns.push({
+        headerName: '',
+        checkboxSelection: true,
+        headerCheckboxSelection: true,
+        width: 28,
+        pinned: 'left',
+        lockPosition: true,
+        suppressMenu: true,
+        suppressSorting: true,
+        suppressFilter: true,
+        suppressResize: true,
+        suppressMovable: true
+      });
+    }
+
+    let index = -1;
+
+    Object.keys(props).forEach(key => {
+      if (attributeBlacklist.includes(key)) {
+        return;
+      }
+
+      if (props[key] instanceof OlGeomGeometry) {
+        return;
+      }
+
+      if (columnDefs[key] && columnDefs[key].index !== undefined) {
+        index = columnDefs[key].index;
+      } else {
+        ++index;
+      }
+      columns[index] = {
+        headerName: key,
+        field: key,
+        ...columnDefs[key]
+      };
+    });
+
+    return columns;
+  }
+
+  /**
+   * Returns the table row data from all of the given features.
+   *
+   * @return {Array} The table data.
+   */
+  getRowData = () => {
+    const {
+      features
+    } = this.props;
+
+    let data = [];
+
+    features.forEach(feature => {
+      const properties = feature.getProperties();
+      const filtered = Object.keys(properties)
+        .filter(key => !(properties[key] instanceof OlGeomGeometry))
+        .reduce((obj, key) => {
+          obj[key] = properties[key];
+          return obj;
+        }, {});
+
+      data.push({
+        key: this.props.keyFunction(feature),
+        ...filtered
+      });
+    });
+
+    return data;
+  }
+
+  /**
+   * Returns the correspondig feature for the given table row key.
+   *
+   * @param {Number} key The row key to obtain the feature from.
+   * @return {ol.Feature} The feature candidate.
+   */
+  getFeatureFromRowKey = key => {
+    const {
+      features,
+      keyFunction
+    } = this.props;
+
+    const feature = features.filter(feature => keyFunction(feature) === key);
+
+    return feature[0];
+  }
+
+  /**
+   * Returns the correspondig rowNode for the given feature id.
+   *
+   * @param {Number} id The feature's unique id to obtain the row from.
+   * @return {Object} The row candidate.
+   */
+  getRowFromFeatureId = id => {
```

Comment:
This function has to be renamed/-factored. As the `getRowData` function makes use of the `keyFunction` prop `node.data.key` will no always contain the id.
A more general function `getRowFromFeature` which makes use of the `keyFunction` prop aswell will do the trick.

Reply:
I've renamed the function, actually it was only used with the key as the argument.